docs/mvvm/MigratingFromMvvmLight.md

```diff
@@ -0,0 +1,791 @@
+
+```csharp
+// MvvmLight
+Messenger.Default.Register<MyMessage>(this, this.OnMyMessageReceived);
+
+// Toolkit.Mvvm
+Messenger.Default.Register<MyMessage>(this, this.OnMyMessageReceived);
+```
+
+#### Register<TMessage> ( object, bool, Action<TMessage> )
+
+There is no direct replacement for this registration mechanism which allows you to support receiving messages for derived message types also. This change is intentional as the `Messenger` implementation aims to not use reflection to achieve it's performance benefits.
+
+Alternatively, there are a few options that can be done to achieve this functionality. 
+
+- Create a custom `IMessenger` implementation
+- Register the additional message types using a shared `Action`
+
+```csharp
+// MvvmLight
+Messenger.Default.Register<MyMessage>(this, true, this.OnMyMessageReceived);
+
+// Toolkit.Mvvm
+Messenger.Default.Register<MyMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+Messenger.Default.Register<MyOtherMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+```
+
+#### Register<TMessage> ( object, object, Action<TMessage> )
+
+The functionality of `Register<TMessage>(object, object, Action<TMessage>)` can be achieved with the MVVM Toolkit's `Register<TMessage, TToken>(object, TToken, Action<TMessage>)` method. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Register<MyMessage>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+
+// Toolkit.Mvvm
+Messenger.Default.Register<MyMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+```
+
+#### Register<TMessage> ( object, object, bool, Action<TMessage> )
+
+There is no direct replacement for this registration mechanism which allows you to support receiving messages for derived message types also. This change is intentional as the `Messenger` implementation aims to not use reflection to achieve it's performance benefits.
+
+Alternatively, there are a few options that can be done to achieve this functionality. 
+
+- Create a custom `IMessenger` implementation
+- Register the additional message types using a shared `Action`
+
+```csharp
+// MvvmLight
+Messenger.Default.Register<MyMessage>(this, nameof(MyViewModel), true, this.OnMyMessageReceived);
+
+// Toolkit.Mvvm
+Messenger.Default.Register<MyMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+Messenger.Default.Register<MyOtherMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+```
+
+#### Send<TMessage> ( TMessage )
+
+The functionality of `Send<TMessage>(TMessage)` can be achieved with the MVVM Toolkit's `IMessenger` extension method `Send<TMessage>(TMessage)`. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Send<MyMessage>(new MyMessage());
+Messenger.Default.Send(new MyMessage());
+
+// Toolkit.Mvvm
+Messenger.Default.Send<MyMessage>(new MyMessage());
+Messenger.Default.Send(new MyMessage());
```

Comment:
We might also want to mention that there are extensions to simplify the syntax when the messages being sent have a parameterless constructor. In that case the message doesn't need to be explicitly created:
```csharp
Messenger.Default.Send<MyMessage>();
```
In this extension that `TMessage` parameter has the `new()` constraint, so it can be instantiated automatically 👍

Reply:
Added the info on this one. I think it's unlikely that messages are being sent without parameters but I like the extension and called it out.